### PR TITLE
fix show annotations

### LINKED ISSUE #1618: The properties table for "vela show annotations" is empty
**Describe the bug**

Applying [annotation trait](https://github.com/oam-dev/kubevela/blob/3be63bba91117272b2704953d2255d73ddbd304f/charts/vela-core/templates/defwithtemplate/annotations.yaml), "vela show annotations" display none of properties.

```
vela show annotations
# Properties
+------+-------------+------+----------+---------+
| NAME | DESCRIPTION | TYPE | REQUIRED | DEFAULT |
+------+-------------+------+----------+---------+
+------+-------------+------+----------+---------+
```

### PR DESCRIPTION
**Pull request by wangyuan249**
###New
Only fix map type parameter (the resolution for cue parameter need to be struct kind ), for  single string , int and bool type  we need to fix in  pkg/cue/convert.go  and add  non-struct kind logic.

We should declare the parameter name and ”+usage"  in cue template
fix #1618 
image
the resolution for cue template in function GetCUEParameterValue has no problem

example for two parameter declare and resolution：
image

### CODE REVIEW

**Comment by wangyuan249** [comment]
Actually the change can not pass "make reviewable",  after make reviewable,  it will degenerate back to the original cue template..  I am looking for a better solution.

File: charts/vela-core/templates/defwithtemplate/annotations.yaml [hunk 1]
```diff
@@ -8,16 +8,19 @@ metadata:
...
-        		for k, v in parameter {
+        		for k, v in parameter.annotations {
         			"\(k)": v
         		}
         	}
         }
-        parameter: [string]: string
+        parameter:
+            // +usage= Specify annotations for your Workload
+            annotations: [string]: string
```
Comment (zzxwill) [inline charts/vela-core/templates/defwithtemplate/annotations.yaml:25]: Why does parameter  `annotations` have to be added?

File: e2e/plugin/plugin_suit_test.go [hunk 2]
```diff
@@ -96,6 +98,16 @@ var _ = BeforeSuite(func(done Done) {
...
+	traitDefCueMap, err := ioutil.ReadFile("../../charts/vela-core/templates/defwithtemplate/annotations.yaml")
+	Expect(err).Should(BeNil())
+	traitDefCueMapStr := string(traitDefCueMap)
+	traitDefCueMapStr = strings.Replace(traitDefCueMapStr, "{{.Values.systemDefinitionNamespace}}", "default", -1)
+
+	Expect(yaml.Unmarshal([]byte(traitDefCueMapStr), &testTdDefCueMap)).Should(BeNil())
+	testTdDefCueMap.Namespace = "default"
+	err = k8sClient.Create(ctx, &testTdDefCueMap)
+	Expect(err).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+
```
Comment (yangsoon) [inline e2e/plugin/plugin_suit_test.go:110]: it seems no need, e2e environment already has `annotations` trait

**Comment by leejanee** [comment]
LGTM

**Pull request merged**